2014-04-05 Kai Pastor (OpenOrienteering)
- Deal with unset QMAKE_LIBDIR_OPENGL when cross-compiling for win32-g++
- cf. QTBUG-34940

--- Qt5/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in.orig	2014-04-05 16:58:58.534843689 +0200
+++ Qt5/qtbase/src/gui/Qt5GuiConfigExtras.cmake.in	2014-04-05 17:17:49.042836735 +0200
@@ -95,12 +95,18 @@ macro(_qt5gui_find_extra_libs Name Libs
     foreach(_lib ${Libs})
         string(REGEX REPLACE "[^_A-Za-z0-9]" "_" _cmake_lib_name ${_lib})
         if (NOT TARGET Qt5::Gui_${_cmake_lib_name})
+            set(_find_library_options)
+            if (LibDir)
+                set(_find_library_options
+                    PATHS \"${LibDir}\"
+!!IF !mac
+                    NO_DEFAULT_PATH
+!!ENDIF
+                )
+            endif()
             find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY ${_lib}
 !!IF !isEmpty(CROSS_COMPILE)
-                PATHS \"${LibDir}\"
-!!IF !mac
-                NO_DEFAULT_PATH
-!!ENDIF
+                ${_find_library_options}
 !!ENDIF
             )
 !!IF mac
@@ -126,10 +132,7 @@ macro(_qt5gui_find_extra_libs Name Libs
             unset(Qt5Gui_${_cmake_lib_name}_LIBRARY CACHE)
 
             find_library(Qt5Gui_${_cmake_lib_name}_LIBRARY_DEBUG ${_lib}d
-                PATHS \"${LibDir}\"
-!!IF !mac
-                NO_DEFAULT_PATH
-!!ENDIF
+                ${_find_library_options}
             )
             if (Qt5Gui_${_cmake_lib_name}_LIBRARY_DEBUG)
                 set_property(TARGET Qt5::Gui_${_cmake_lib_name} APPEND PROPERTY IMPORTED_CONFIGURATIONS DEBUG)
